From 6119ca5f113320d848ead38b141238b013f538c6 Mon Sep 17 00:00:00 2001
From: Laurentiu Tudor <laurentiu.tudor@nxp.com>
Date: Wed, 23 Jan 2019 16:28:36 +0200
Subject: [PATCH] sdk: dts: ls104x: drop smmu from the qds and usdpaa sdk dts

Also drop the smmu from the qds and usdpaa versions of the
SDK device trees because SMMU is supported only for the
upstream version of the dpaa ethernet drivers.

Signed-off-by: Laurentiu Tudor <laurentiu.tudor@nxp.com>
---
 arch/arm64/boot/dts/freescale/fsl-ls1043a-qds-sdk.dts    | 14 ++++++++++++++
 arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-usdpaa.dts | 14 ++++++++++++++
 arch/arm64/boot/dts/freescale/fsl-ls1046a-qds-sdk.dts    | 14 ++++++++++++++
 arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-usdpaa.dts | 14 ++++++++++++++
 4 files changed, 56 insertions(+)

--- a/arch/arm64/boot/dts/freescale/fsl-ls1043a-qds-sdk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1043a-qds-sdk.dts
@@ -64,6 +64,20 @@
 &soc {
 #include "qoriq-dpaa-eth.dtsi"
 #include "qoriq-fman3-0-6oh.dtsi"
+
+pcie@3400000 {
+       /delete-property/ iommu-map;
+};
+
+pcie@3500000 {
+       /delete-property/ iommu-map;
+};
+
+pcie@3600000 {
+       /delete-property/ iommu-map;
+};
+
+/delete-node/ iommu@9000000;
 };
 
 &fman0 {
--- a/arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-usdpaa.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-usdpaa.dts
@@ -92,6 +92,20 @@
 			fsl,fman-oh-port = <&fman0_oh2>;
 		};
 	};
+
+	pcie@3400000 {
+	       /delete-property/ iommu-map;
+	};
+
+	pcie@3500000 {
+	       /delete-property/ iommu-map;
+	};
+
+	pcie@3600000 {
+	       /delete-property/ iommu-map;
+	};
+
+	/delete-node/ iommu@9000000;
 };
 / {
 	reserved-memory {
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds-sdk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds-sdk.dts
@@ -64,6 +64,20 @@
 &soc {
 #include "qoriq-dpaa-eth.dtsi"
 #include "qoriq-fman3-0-6oh.dtsi"
+
+pcie@3400000 {
+       /delete-property/ iommu-map;
+};
+
+pcie@3500000 {
+       /delete-property/ iommu-map;
+};
+
+pcie@3600000 {
+       /delete-property/ iommu-map;
+};
+
+/delete-node/ iommu@9000000;
 };
 
 &fsldpaa {
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-usdpaa.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-usdpaa.dts
@@ -85,6 +85,20 @@
 			fsl,fman-oh-port = <&fman0_oh2>;
 		};
 	};
+
+	pcie@3400000 {
+	       /delete-property/ iommu-map;
+	};
+
+	pcie@3500000 {
+	       /delete-property/ iommu-map;
+	};
+
+	pcie@3600000 {
+	       /delete-property/ iommu-map;
+	};
+
+	/delete-node/ iommu@9000000;
 };
 / {
 	reserved-memory {
